pkg/solana/chainwriter/chain_writer.go

```diff
@@ -316,10 +390,41 @@ func (s *SolanaChainWriterService) SubmitTransaction(ctx context.Context, contra
 	discriminator := GetDiscriminator(methodConfig.ChainSpecificName)
 	encodedPayload = append(discriminator[:], encodedPayload...)
 
+	// Fetch latest blockhash
+	blockhash, err := s.client.LatestBlockhash(ctx)
+	if err != nil {
+		return errorWithDebugID(fmt.Errorf("error fetching latest blockhash: %w", err), debugID)
+	}
+
+	if len(createATAinstructions) > 0 {
+		ataTx, ataErr := solana.NewTransaction(
+			createATAinstructions,
```

Comment:
Not sure about the worst case scenario but can we can safely just pass all the instructions into 1 transaction and not worry about tx size? If we're not sure and we know the avg size of an ATA instruction, we can maybe set a batch limit

Reply:
After some testing, I've concluded that 12 ATAs is the maximum number of batch create ATA instructions that can be included in a single transaction. I've added a check that ensures the length of the `createATAinstructions` doens't exceed this.

Reply:
Maybe we should double check with on-chain if it's a valid scenario to need more than 12 ATA instructions. If it is, we may want to allow this but just batch them into different transactions. These we could probably fire all at the same time. Unless we want to keep a limit anyways to cap costs but regardless think we should double check with on-chain.

Reply:
Yes, batching is preferred.
If you don't want to do that right away, atleast instead of failing this check 
"if len(createATAinstructions) > maxAtas",
Tt is better to cap ATAs to maxAtas.
This way, we create 12 ATAs only, and ognore others. This will mean the original Tx will too fail, but then we have other nodes in the DON, which will pick this up, and create the remaining ATAs and the CCIP message won't get blocked.

Another workaround is to get confirmation from CCIP team that 12 token transfers in 1 message isn't ever going to happen.

Reply:
From Agustina's comment in the slack thread, it seems that just failing here is the best option. The execution report will not go through with so many accounts, so there's no point wasting the gas to create the ATAs.

Reply:
Based on offline discussions, sounds like >12 ATA instructions isn't possible so I'm ok with keeping this as an error.